docs/source/cuda/cudapysupported.rst

```diff
@@ -54,6 +56,44 @@ This is due to a general limitation in CUDA printing, as outlined in the
 <https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#limitations>`_
 in the CUDA C++ Programming Guide.
 
+
+Recursion
+---------
+
+Self-recursive device functions are supported, with the constraint that
+recursive calls must all have similar argument types. For example, the following
```

Comment:
I've adjusted the wording - I tried to avoid the word "signature" as I thought that might not have a clear meaning to the average reader, but hopefully it conveys the required meaning now.